platform-coop-toolkit/coop-library-framework
# Add unique identifier fields and add publication year (resolves #86)

## Description

Adds fields for:

- [DOI](https://www.doi.org/)
- [ISBN](https://en.wikipedia.org/wiki/International_Standard_Book_Number)
- [ISSN](https://en.wikipedia.org/wiki/International_Standard_Serial_Number)

Adds validation for input to these fields.

Add publication year field.

Renames base plugin file name.

## Steps to test

1. Reactivate plugin (base file name has changed).
2. Add invalid DOI, ISBN, and ISSN values; add invalid publication year.
3. Attempt save; observe validation errors.
4. Add valid DOI, ISBN, and ISSN values; add publication year.
5. Save.

**Expected behavior:**

* Invalid DOI, ISBN, ISSN, and publication year values should warn the user of formatting issues and prevent form submission.
* Valid DOI, ISBN, ISSN, and publication year values should save.

## Additional information

Depends on #85.

## Related issues

- Resolves #86
